lib/net.js

```diff
@@ -299,6 +302,10 @@ Socket.prototype.listen = function() {
 
 
 Socket.prototype.setTimeout = function(msecs, callback) {
+  if (typeof msecs === 'function') {
```

Comment:
How is this related to PR / why was it added? Is there a case where we want to allow to omit time?